### PR TITLE
Introducing Store Police

### PR DESCRIPTION
**Pull request by giograno**
Store Police is a defensive mechanism against pickling errors. In a nutshell, after each test, it attempts at pickling the backends for the running services to make sure they are serializable.

### CODE REVIEW

File: localstack/testing/pytest/persistence.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    result: PicklingTestResult
+
+    def __init__(self, result: PicklingTestResult):
+        super().__init__()
+        self.result = result
+
+
+class DirtyMarkerHandler(Handler):
+    """
+    A handler injected into the handler chain to only
```
Comment (thrau) [inline localstack/testing/pytest/persistence.py:94]:
```suggestion
    A handler injected into the handler chain to mark services that have been used by a test, to make sure we only make a serialization check for those.
```

**Review by thrau — COMMENTED** [review]
mechanism looks great! it's great to see how easily we can add these types of mechanisms with the stores and visitors!

i suggest moving this into localstack-ext though, so we don't create the circular dependency between localstack-core and localstack-plugin-persistence :+1:

**Comment by giograno** [comment]
> i suggest moving this into localstack-ext though, so we don't create the circular dependency between localstack-core and localstack-plugin-persistence 👍

yeah, I was thinking the same. Especially considering the upcoming workstreams. Going to open a PR in ext.

**Comment by giograno** [comment]
Closing in favor of https://github.com/localstack/localstack-ext/pull/1622.

**Pull request closed without merging**